cedar-policy/src/tests.rs

```diff
@@ -3258,3 +3258,31 @@ mod error_source_tests {
         }
     }
 }
+
+mod issue_618 {
```

Comment:
I think it's worth being exhaustive here. These extra test cases come to mind.

* `"" like "\n"`
* `principal has "\n"`
* `principal["\n"]`
* `{"\n": 0}`
* `@annotation("\n")`

Reply:
+1

Reply:
could still use these test cases